src/pages/settings/Preferences/LanguagePage.js

```diff
@@ -0,0 +1,75 @@
+import _ from 'underscore';
+import React from 'react';
+import HeaderWithCloseButton from '../../../components/HeaderWithCloseButton';
+import ScreenWrapper from '../../../components/ScreenWrapper';
+import withLocalize, {withLocalizePropTypes, localeProviderPropTypes} from '../../../components/withLocalize';
+import Navigation from '../../../libs/Navigation/Navigation';
+import ROUTES from '../../../ROUTES';
+import OptionsList from '../../../components/OptionsList';
+import styles from '../../../styles/styles';
+import themeColors from '../../../styles/themes/default';
+import * as Expensicons from '../../../components/Icon/Expensicons';
+import * as App from '../../../libs/actions/App';
+
+const greenCheckmark = {src: Expensicons.Checkmark, color: themeColors.success};
+
+const propTypes = {
+    ...withLocalizePropTypes,
+    ...localeProviderPropTypes,
+};
+
+const LanguagePage = (props) => {
+    const localesToLanguages = _.map(props.translate('languagePage.languages'),
+        (language, key) => (
+            {
+                value: key,
+                text: language.label,
+                keyForList: key,
+
+                // Include the green checkmark icon to indicate the currently selected value
+                customIcon: props.preferredLocale === key ? greenCheckmark : undefined,
+
+                // This property will make the currently selected value have bold text
+                boldStyle: props.preferredLocale === key,
+            }
+        ));
+
+    return (
+        <ScreenWrapper includeSafeAreaPaddingBottom={false}>
+            <HeaderWithCloseButton
+                title={props.translate('languagePage.language')}
+                shouldShowBackButton
+                onBackButtonPress={() => Navigation.navigate(ROUTES.SETTINGS_PREFERENCES)}
+                onCloseButtonPress={() => Navigation.dismissModal(true)}
+            />
+            <OptionsList
+                sections={[{data: localesToLanguages}]}
+                onSelectRow={
+                    (language) => {
+                        if (language.value !== props.preferredLocale) {
+                            App.setLocale(language.value);
+                        }
+                        Navigation.navigate(ROUTES.SETTINGS_PREFERENCES);
+                    }
+                }
+                hideSectionHeaders
+                optionHoveredStyle={
+                    {
+                        ...styles.hoveredComponentBG,
+                        ...styles.mln5,
+                        ...styles.mrn5,
+                        ...styles.ph5,
+                    }
+                }
```

Comment:
```suggestion
                optionHoveredStyle={[
                    styles.hoveredComponentBG,
                    styles.mln5,
                    styles.mrn5,
                    styles.ph5,
                ]}
```

Reply:
I tried this from the first place but the thing is that optionHoveredStyle is a custom prop and its type is object. There are other places in the App where it is used as an object. Do you think I should make it as an array and change it everywhere?

Reply:
Hmm true....... Pretty much all style propsTypes we have are **either** `style: PropTypes.arrayOf(PropTypes.object),` OR they allow arrays-of-objects OR plain objects. This is one of the last places we **only** allow objects in styles, so.........

I would preferrrrrr we update that style PropType to be an array of objects OR a plain object - what do you think?

Reply:
simplified, but I wouldn't change the type of this prop in this PR. It impacts multiple components and I think we need rigorous testing.